routers/web/repo/githttp.go

```diff
@@ -196,8 +198,22 @@ func httpBase(ctx *context.Context) *serviceHandler {
 					return nil
 				}
 				if task.RepoID != repo.ID {
-					ctx.PlainText(http.StatusForbidden, "User permission denied")
-					return nil
+					taskRepo, exist, err := db.GetByID[repo_model.Repository](ctx, task.RepoID)
```

Comment:
Is write access needed? This could be abused, by workflow code referencing ${{ gitea.token }} to modify action code.

```suggestion
					if accessMode > perm.AccessModeRead {
						ctx.PlainText(http.StatusForbidden, "User permission denied")
						return nil
					}
					taskRepo, exist, err := db.GetByID[repo_model.Repository](ctx, task.RepoID)
```

~~What about PRs from fork? Should they get read access?, They could reupload your private repos, if you ignore the workflow approval protection.~~

~~_I think that having this feature for public repos would be a big advantage over GitHub Actions, while they do not grant github.token this power and use other, inaccessible for user code except ram dumps, tokens inside the runner for this purpose_~~

**EDIT** Sounds like private repo only, like GitHub. Large parts are invalidated